doc/api/apollo-voyager-server.md

```diff
@@ -0,0 +1,139 @@
+
+This options object is used to enable the additional features provided by the Voyager framework. 
+
+* `VoyagerConfig`: <`Object`>
+
+  * `securityService`: <[`SecurityService`](#SecurityService)> - A security service module that can add authentication and authorization to a voyager application. See the [apollo-voyager-keycloak] module as an example.
+
+  ```js
+    const { KeycloakSecurityService } = require('@aerogear/apollo-voyager-keycloak')
+
+    const keycloakConfig = JSON.parse(fs.readFileSync( './config/keycloak.json')))
+
+    // initialize the security service
+    const securityService = new KeycloakSecurityService(keycloakConfig)
+
+    const apolloConfig = { typeDefs, resolvers }
+    const voyagerConfig = { securityService }
+
+    const server = ApolloVoyagerServer(apolloConfig, voyagerConfig)
+
+    const app = express()
+
+    // call the apply the security service middleware to your express app
+    // optionally takes a second config argument like { apiPath: '/mygraphqlendpoint' }
+    securityService.applyAuthMiddleware(app)
+    server.applyMiddleware({ app })
+
+    app.listen(4000, () =>
+      console.log(`🚀 Server ready at http://localhost:4000/graphql`)
+    )
+  ```
+
+  * `metrics`: <[`Metrics`](#Metrics)> - A metrics module that can add resolver level metrics and metrics middlewares to your application. See the [apollo-voyager-metrics] module.
+
+
+  ```js
+  const metrics = require('@aerogear/apollo-voyager-metrics')
+
+  const apolloConfig = { typeDefs, resolvers }
+  const voyagerConfig = { metrics }
+
+  const server = ApolloVoyagerServer(apolloConfig, voyagerConfig)
+
+  const app = express()
+
+  // apply the metrics middlwares to the express app
+  metrics.applyMetricsMiddlewares(app)
+
+  server.applyMiddleware({ app })
+
+  app.listen(4000, () =>
+    console.log(`🚀 Server ready at http://localhost:4000/graphql`)
+  )
+  ```
+
+  * `auditLogger`: <[`AuditLogger`](#AuditLogger)> - An Audit Logger module that can add Audit Logging to your application for GraphQL operations, authentication and authorization operations, conflicts, etc. See the [apollo-voyager-audit] module.
+
+  ```js
+  const auditLogger = require('@aerogear/apollo-voyager-audit')
+
+  const apolloConfig = { typeDefs, resolvers }
+  const voyagerConfig = { auditLogger }
+
+  const server = ApolloVoyagerServer(apolloConfig, voyagerConfig)
+
+  const app = express()
+
+  server.applyMiddleware({ app })
+
+  app.listen(4000, () =>
+    console.log(`🚀 Server ready at http://localhost:4000/graphql`)
+  )
+  ```
+
+### SecurityService
+
+See the [SecurityService] interface.
+
+### Metrics
+
+See the [Metrics] interface.
+
+### AuditLogger
+
+See the [AuditLogger] interface.
```

Comment:
[Question] Do we need separate tasks to cover those?

Reply:
The main reason someone would want to know about these interfaces is if they wanted to build their own metrics/audit logging implementations. In my opinion (you may disagree) we shouldn't be encouraging anyone to build custom implementations just yet.

I figure that it is so early, the specifics of these interfaces are likely to change again. Instead of documenting them I have linked directly to the interface definitions. It's not perfect but the links to the definitions are better than nothing.